pkg/initializer/initializer.go

```diff
@@ -42,6 +46,42 @@ import (
 //		   - No snapshots are available, start etcd as a fresh installation.
 func (e *EtcdInitializer) Initialize(mode validator.Mode, failBelowRevision int64) error {
 	start := time.Now()
+	if !e.IsMemberInCluster() {
+
+		//Add member as learner to cluster
+		memAddCtx, cancel := context.WithTimeout(context.TODO(), 5*time.Second)
+		defer cancel()
+		memberURL := getMemberURL()
+		if memberURL == "" {
+			e.Logger.Warn("Could not fetch member URL")
+		}
+		for {
+			//Create etcd client
+			clientFactory := etcdutil.NewFactory(brtypes.EtcdConnectionConfig{
+				Endpoints:         []string{"http://etcd-main-peer.default.svc:2380"}, //TODO: use ETCD_ENDPOINT env var passed by druid
+				InsecureTransport: true,
```

Comment:
Whenever we want to connect to `embedded etcd` or its corresponding `etcd` we set `InsecureTransport: true` to get etcd client.
But here you need to connect to a etcd member of etcd cluster running in a different pod to get etcd client then I guess you should try to have secure connection. WDYT?